### PR TITLE
Replace linter with jsonschema validation using the SHARE schema

### PR DESCRIPTION
**Pull request by fabianvf**
Welp, everything looks pretty much good. Just need to wait for the final schema now (and change stuff on the OSF side)

Migration steps:

```local.py: ELASTIC_INDEX = 'share_v2'```
```inv reindex share share_v1 && inv alias share share_v1 && inv renormalize```

And once the OSF is up to date with scrAPI:
```inv alias share share_v2```

TODO
---------
- [x] arxiv.py
- [x] asu.py
- [x] biomed.py
- [x] calpoly.py
- [x] clinicaltrials.py
- [x] cmu.py
- [x] columbia.py
- [x] crossref.py
- [x] dataone.py
- [x] doepages.py
- [x] figshare.py
- [x] mit.py
- [x] opensiuc.py
- [x] plos.py
- [x] pubmed.py
- [x] scitech.py
- [x] spdataverse.py
- [x] stcloud.py
- [x] tdar.py
- [x] texasstate.py
- [x] trinity.py
- [x] ucescholarship.py
- [x] uiucideals.py
- [x] upennsylvania.py
- [x] utaustin.py
- [x] uwashington.py
- [x] valposcholar.py
- [x] vtech.py
- [x] waynestate.py
- [x]  Database models
- [x] Tests
- [x] Migration

### CODE REVIEW

**Comment by fabianvf** [comment]
Working on Biomed now

**Comment by fabianvf** [comment]
Moving on to OAI

File: scrapi/base/__init__.py [hunk 1]
```diff
@@ -115,7 +86,8 @@ class XMLHarvester(BaseHarvester, XMLTransformer):
 
     def normalize(self, raw_doc):
         transformed = self.transform(etree.XML(raw_doc['doc']))
-        transformed['source'] = self.short_name
+        transformed['shareProperties'] = {}
+        transformed['shareProperties']['source'] = self.short_name
```
Comment (chrisseto) [inline scrapi/base/__init__.py]: Why not:
```
transformed['shareProperties'] = {
    'source': self.short_name
}
```

File: scrapi/harvesters/crossref.py [hunk 2]
```diff
@@ -26,13 +26,12 @@
 def process_contributor(author, orcid):
     name = HumanName(author)
     return {
-        'prefix': name.title,
-        'given': name.first,
-        'middle': name.middle,
-        'family': name.last,
-        'suffix': name.suffix,
+        'name': author,
```
Comment (chrisseto) [inline scrapi/harvesters/crossref.py]: Any reason why this isn't a helper?
It looks like its used in a couple other places.

File: scrapi/processing/elasticsearch.py [hunk 3]
```diff
@@ -21,14 +21,15 @@
...
-    }
-    es.cluster.health(wait_for_status='yellow')
-    es.indices.create(index=settings.ELASTIC_INDEX, body=body, ignore=400)
+    # body = {
+    #     'mappings': {
+    #         harvester: settings.ES_SEARCH_MAPPING
+    #         for harvester in registry.keys()
+    #     }
+    # }
+    # es.cluster.health(wait_for_status='yellow')
```
Comment (chrisseto) [inline scrapi/processing/elasticsearch.py]: Why did this get commented out?

File: scripts/renormalize.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                        'doc': doc.doc,
+                        'docID': doc.docID,
+                        'source': doc.source,
+                        'filetype': doc.filetype,
+                        'timestamps': doc.timestamps
+                    })
+                except Exception as e:
+                    print(e)
+        page = list(query.filter(pk__token__gt=Token(page[-1].pk)))
+    print(count)
```
Comment (chrisseto) [inline scripts/renormalize.py]: ?

File: scripts/renormalize.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+
+for raw in document_generator():
```
Comment (chrisseto) [inline scripts/renormalize.py]: Guard this with 
```python
def main():
    ...


if __name__ == '__main__':`
    main()

File: scripts/renormalize.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    try:
+        process_normalized(normalize(raw, raw['source']), raw)
+    except Exception as e:
+        print(e)
```
Comment (chrisseto) [inline scripts/renormalize.py]: Prefer use of a logger

File: scrapi/base/__init__.py (hunk 1 above)
Comment (fabianvf) [inline scrapi/base/__init__.py]: Because I am dumb.

File: scrapi/harvesters/crossref.py (hunk 2 above)
Comment (fabianvf) [inline scrapi/harvesters/crossref.py]: process_contributors is used in a lot of places, but I think this is the only harvester that processes one contributor at a time.

File: scrapi/processing/elasticsearch.py (hunk 3 above)
Comment (fabianvf) [inline scrapi/processing/elasticsearch.py]: Not sure what we want the new mappings to be yet. Will probably replace this with something that generates mappings from a jsonschema specification.

File: scripts/renormalize.py (hunk 4 above)
Comment (fabianvf) [inline scripts/renormalize.py]: Just wanted to make sure it iterated over the correct number of documents. Can be deleted.

File: scripts/renormalize.py (hunk 5 above)
Comment (fabianvf) [inline scripts/renormalize.py]: Good idea.

File: scripts/renormalize.py (hunk 6 above)
Comment (fabianvf) [inline scripts/renormalize.py]: mos def.

File: scrapi/base/__init__.py (hunk 1 above)
Comment (chrisseto) [inline scrapi/base/__init__.py]: *rushed

**Pull request merged**